### PR TITLE
Fix postmerge errors

### PR DESCRIPTION
**Pull request by mattiasostmar**
Task: https://phabricator.wikimedia.org/T164715

### CODE REVIEW

File: create_infotexts.py [hunk 1]
```diff
@@ -494,13 +489,13 @@ def process_region_addition_to_description(region_str, country_str):
...
 
-    def add_catch_all_category(self, item):
+    def add_catch_all_category(self):
         """"
         Check if there are any content cats added and add generic content category to image.
         
         Populate self.content_cat with commons category, if present.
         """
-        if not self.content_cats:
+        if len(self.content_cats) == 0:
```
Comment (lokal-profil) [inline create_infotexts.py]: Why this change? The previous one should catch both empty list and `None`

File: create_infotexts.py [hunk 2]
```diff
@@ -371,7 +371,11 @@ def process_depicted_place(self, place_string, places_mapping, desc_string):
         place_as_wikitext = ""
 
         if place_string:
-            if place_string in places_mapping:
+            # Don't forget to add the commons categories, even though only wikidata is used in depicted people field
+            if places_mapping[place_string].get('commonscat'):
```
Comment (lokal-profil) [inline create_infotexts.py]: This re-jig doesn't work.

In the previous version we first checked that place_string  was in places_mapping. If this was the case we then also checked for a commonscat. Here `places_mapping[place_string]` is likely to throw a `keyError`.

To fix re-add `if place_string in places_mapping:` before this (see also next comment).

File: create_infotexts.py [hunk 3]
```diff
@@ -371,7 +371,11 @@ def process_depicted_place(self, place_string, places_mapping, desc_string):
...
+                self.content_cats.append(places_mapping[place_string]["commonscat"])
+
+            if place_string.lower() in [place.lower() for place in places_mapping.keys()]:
```
Comment (lokal-profil) [inline create_infotexts.py]: Is there really a mixture of upper/lower in both the mapping and the in-data?

If so I would standardise the mapping data (just make them all lower). Then in the `if place_string in places_mapping` above (see previous comment) simply change `place_string` to `place_string.lower()` (and drop this if).

File: create_infotexts.py [hunk 4]
```diff
@@ -392,18 +396,18 @@ def process_depicted_place(self, place_string, places_mapping, desc_string):
...
                 place_as_wikitext += place_string
 
+
         else:
             place_matches = []
             for place in places_mapping:
                 if place.lower() in desc_string.lower():
                     place_matches.append(place)
+                    # Don't forget to add the commons categories if present.
+                    if places_mapping[place].get('commonscat'):
```
Comment (lokal-profil) [inline create_infotexts.py]: Don't move this up from [L415](https://github.com/mattiasostmar/SMVK-Cypern_2017-01/pull/35/files#diff-7dcfcee85feab8260c18d5fc3b799639L415). This was on purpose inside the `if len(place_matches) == 1:` to ensure we only add the category on an exact match.

That said I noticed that on the line before [L409](https://github.com/mattiasostmar/SMVK-Cypern_2017-01/pull/35/files#diff-7dcfcee85feab8260c18d5fc3b799639L409) we should have introduced: `place=place_matches[0]`.

The commonscat on [L423](https://github.com/mattiasostmar/SMVK-Cypern_2017-01/pull/35/files#diff-7dcfcee85feab8260c18d5fc3b799639L423) was correctly removed.

**Review by lokal-profil — CHANGES_REQUESTED** [review]

File: create_infotexts.py (hunk 1 above)
Comment (mattiasostmar) [reply create_infotexts.py]: I changed because I tried this:
    In [25]: l = []

    In [26]: l == True
    Out[26]: False

    In [27]: l is True
    Out[27]: False

    In [28]: l == False
    Out[28]: False

    In [29]: l is False
    Out[29]: False
However, quirky enough the following works:
 ```
 In [30]: if l:
              print("l is True?")
          else:
              print("nah, l is False.")
nah, l is False.
```

Comment (lokal-profil) [reply create_infotexts.py]: That is because if doesn't just test for True/False but (in this case also interprets `None`,`[]`,`{}`,`0` and likely other empty elements as False.

The only one of these that might be very dangerous is `0`. I.e. if you are evaluating numbers and `num=0` is a valid number then you cannot use `if num` to guard agains `num is None`.

But that is not the case here so `if not self.content_cats:` should be used.

Comment (mattiasostmar) [reply create_infotexts.py]: Done

**Review by lokal-profil — APPROVED** [review]
